src/main/java/com/google/devtools/build/lib/remote/RemoteExecutionService.java

```diff
@@ -375,23 +376,29 @@ public RemoteAction buildRemoteAction(Spawn spawn, SpawnExecutionContext context
   public static class RemoteActionResult {
     private final ActionResult actionResult;
     @Nullable private final ExecuteResponse executeResponse;
+    private final String cacheName;
 
     /** Creates a new {@link RemoteActionResult} instance from a cached result. */
     public static RemoteActionResult createFromCache(ActionResult cachedActionResult) {
+      return createFromCache(cachedActionResult, "remote");
+    }
+
+    public static RemoteActionResult createFromCache(ActionResult cachedActionResult, String cacheName) {
```

Comment:
Can we change the parameter from `ActionResult` to `CachedActionResult` instead of adding another overload?

Reply:
changed back to original way.
Passing in the CachedActionResult and storing it or pulling out the string felt awkward.
So I changed the two callers to keep the CacheActionResult and the RemoteActionResult